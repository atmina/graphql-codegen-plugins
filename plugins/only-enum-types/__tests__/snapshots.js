const tester = require('@atmina/codegen-tester');
tester('only-enum-types', {
  'documents/__generated__/types.ts': {
    plugins: ['./lib/index.js'],
  }
}).catch(console.error);
