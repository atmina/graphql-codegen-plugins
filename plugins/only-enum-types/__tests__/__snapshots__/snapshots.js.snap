// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The plugin only-enum-types should generate code that matches the snapshots: types.ts 1`] = `
"export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
export type MakeEmpty<T extends { [key: string]: unknown }, K extends keyof T> = { [_ in K]?: never };
export type Incremental<T> = T | { [P in keyof T]?: P extends ' $fragmentName' | '__typename' ? T[P] : never };

export enum Genre {
  Crime = 'CRIME',
  Fantasy = 'FANTASY',
  Thriller = 'THRILLER'
}
"
`;
